docs/advanced_usage.rst

```diff
@@ -0,0 +1,333 @@
+        propagator = MyRPCPropagator()
+        context = propagator.extract(rpc_metadata)
+        tracer.context_provider.activate(context)
+
+        with tracer.trace("child_span") as span:
+            span.set_meta('my_rpc_method', method)
+
+
+Sampling
+--------
+
+.. _`Priority Sampling`:
+
+Priority Sampling
+^^^^^^^^^^^^^^^^^
+
+Priority sampling gives you control over whether or not a trace will be
+propagated. This is done by associating a priority attribute on a trace that
+will be propagated along with the trace. The priority value informs the Agent
+and the backend about how to deal with the trace. By default priorities are set
+on a trace by a sampler.
+
+The sampler can set the priority to the following values:
+
+- ``AUTO_REJECT``: the sampler automatically rejects the trace
+- ``AUTO_KEEP``: the sampler automatically keeps the trace
+
+For now, priority sampling is disabled by default. Enabling it ensures that your
+sampled distributed traces will be complete.  To enable priority sampling::
+
+    tracer.configure(priority_sampling=True)
+
+Once enabled, the sampler will automatically assign a priority to your traces,
+depending on their service and volume.
+
+You can also set this priority manually to either drop an uninteresting trace or
+to keep an important one.
+To do this, set the ``context.sampling_priority`` to one of the following:
+
+- ``USER_REJECT``: the user asked to reject the trace
+- ``USER_KEEP``: the user asked to keep the trace
+
+When not using distributed tracing, you may change the priority at any time, as
+long as the trace is not finished yet.
+But it has to be done before any context propagation (fork, RPC calls) to be
+effective in a distributed context.
+Changing the priority after context has been propagated causes different parts
+of a distributed trace to use different priorities. Some parts might be kept,
+some parts might be rejected, and this can cause the trace to be partially
+stored and remain incomplete.
+
+If you change the priority, we recommend you do it as soon as possible, when the
+root span has just been created::
+
+    from ddtrace.ext.priority import USER_REJECT, USER_KEEP
+
+    context = tracer.context_provider.active()
+
+    # indicate to not keep the trace
+    context.sampling_priority = USER_REJECT
+
+    # indicate to keep the trace
+    span.context.sampling_priority = USER_KEEP
+
+
+Pre-sampling
+^^^^^^^^^^^^
+
+Pre-sampling will completely disable instrumentation of some transactions and
+drop the trace at the client level. Information will be lost but it allows to
+control any potential performance impact.
+
+``RateSampler`` randomly samples a percentage of traces::
+
+    from ddtrace.sampler import RateSampler
+
+    # Sample rate is between 0 (nothing sampled) to 1 (everything sampled).
+    # Keep 20% of the traces.
+    sample_rate = 0.2
+    tracer.sampler = RateSampler(sample_rate)
+
+
+Resolving deprecation warnings
```

Comment:
I'm not sure if this is the best place to put this section or if we even need it.